### ISSUE TITLE
[AZE0] While quickcharging, some stations show incorrect %SOC

### ISSUE THREAD

**Issue by dalathegreat** [dalathegreat/BatteryUpgrade-UserManual#11]
Some quickchargers don't show correct state of charge on battery upgraded LEAFs. Issue is worse on US-imported AZE0s, since these have different ways to communicate capacity.

On some stations this is purely cosmetic issue. For instance, Veefil QC stations will show 100% when the vehicle is actually at ~70%SOC, but this does not matter and charging still continues.

On some other stations this is a more serious issue. For instance, Circontrol branded ones, usually found in Lithuania, the vehicle will stop charging when the station reaches 100% SOC. This leads to the vehicle only getting charged to 70% on these types of station.

No known workaround yet, under investigation. 

Are you affected by this? Feel free to post details about the stations and your vehicle details.

**Comment by nzautomate** [comment]
This issue is a bit of an issue for us, chargenet who operate the fast chargers in out area terminate the charge at either (selectable on veefil station) 80% or 95%. This leaves us unable to 'fast' charge the top portion of the battery if desired

**Comment by dalathegreat** [comment]
I am bumping up prio to high on this issue!

**Comment by glynhudson** [comment]
I've also experienced this issue, resulting in not being able to charge my 40kWh e-NV200 (upgraded from 24kWh) past 67% SoC on DBT Rapid chargers. The rapid charger displayed 100% on the display when the actual SoC was 67% resulting in the charging being stopped. This is a bit awkward since DBT units are the most common rapid charger on UK motorways 

https://photos.app.goo.gl/UKfdukcFgevZz2SN6

Interesting, the issue remained with and without the CAN bridge connected.

**Comment by dalklein** [comment]
Do you still have this issue glyn?  It may be different, but symptom sounds similar to what happened for my 62kwh 2014 conversion, initially in Jan 2020.

It worked after changing the 59e message as below with mux's canbridge code, and Dala's later code had the fix and worked.

Here's what I added to fix for the fast charger SOC,  suggested by mux, this worked.   
            case 0x59E:   // QC capacity message, adjust for 2014 with 2019 62kWhr pack swap
                #ifdef LEAF_2014
                #ifdef BATTERY_SWAP_62
                //   read LBC signals, Full and Remaining capacity
                LBC_QC_FullCap = ((frame.data[2] & 0x1F)<< 4) | ((frame.data[3] & 0xF0) >> 4);
                // LBC_QC_CapR = ((frame.data[3] & 0x0F)<< 5) | ((frame.data[4] & 0xF8) >> 3);   // don't need to read LBC's value
                temp2 = main_battery_soc * LBC_QC_FullCap; //  LBC's QC_FullCap value * pct SOC(0-100)
                temp2 = temp2 / 100;                       //  new LBC_QC_CapR value to report to the charger, based on soc & QC_FullCap
                frame.data[3] = (frame.data[3] & 0xF0) | ((temp2 >> 5) & 0xF);     //     store the new LBC_QC_CapR 
                frame.data[4] = ((temp2 & 0x1F) <<3) | (frame.data[4] & 0x07);     //        to the 59E message out to vehicle
                calc_crc8(&frame);
                #endif
                #endif
                break;    // case 0x59E:

**Comment by dalathegreat** [comment]
Beta firmware dispatched to @glynhudson , let's see if the 0x59E fix helps on the 24kWh env200 VCM!

**Comment by glynhudson** [comment]
Hi @dalathegreat

I've just tested V2.51 FW, it's fix the issue with glide and drive sporadically enabling. However, there have been no change in the rapid charge behaviour. On DBT rapid chargers charging was forced stopped at about 68% due to the rapid charger displaying 100% SoC.

Evbox and ABB rapid chargers work fine, these units done display and percentage and don't seem to force stop a charge.

Is there any data which would be useful for me to collect to help fix this issue? E.g CAN bus log?

**Comment by dalathegreat** [comment]
> Is there any data which would be useful for me to collect to help fix this issue? E.g CAN bus log?

Yes, a log from EV-CAN while quickcharging fails would be great! This can be taken from the OBD2 port (pins 12L&13H)

Glad to hear the env200 glide in drive sporadical activation was fixed atleast :)

**Comment by glynhudson** [comment]
Sure, will do. I can log EV-CAN traffic using OVMS supported formats are: crtd, gvret-a, gvret-b, lawricel, pcap, raw. Do you have a preference? 

https://docs.openvehicles.com/en/latest/crtd/can_logging.html

**Comment by dalathegreat** [comment]
@glynhudson No preference :)

**Comment by dalathegreat** [comment]
AZE0 fix verified working on my end, will be v2.53 shipping out to all AZE0 customers !

**Comment by glynhudson** [comment]
I can confirm this issue is fixed using V2.53 when the CAN bridge is installed in the correct location. See explanation video on why I needed to move the CAN bridge to be closer to the battery on my Nissan e-NV200 https://youtu.be/W-W7FwTNOnU

I think this issue can now be closed, nice work Dala :+1:

**Comment by nzautomate** [comment]
Hi Dala, any progress on a ZE0 fix for this?

**Comment by dalathegreat** [comment]
@nzautomate I am getting a quickcharger installed in my garage next month so I can troubleshoot this in lab-conditions!

**Comment by nzautomate** [comment]
Still waiting to hear if this can be fixed. We will need to change to another Can bridge if you do not plan on fixing it. Now I'm our summer holidays here in NZ and we are going to be taking the ice car where we should be able to manage with the leaf

**Comment by dalathegreat** [comment]
Quickcharging station is ordered (should be in jan2022), electrical system has been upgraded from the ground up to support the 40kW station. Slow (and expensive) progress is made....

**Comment by Toby-TheEnglishMechanic** [comment]
As a current 'status' update, I'd like to add that on my 2012 with 40kwh, my SOC drops as soon as I start the Chademo session, and reads far less than the incoming, whilst charging - clearly not actually an issue, as I can see from the charger how much I've taken on board, but baffles my wife sometimes when she's alone  .....

**Comment by dalathegreat** [comment]
> As a current 'status' update, I'd like to add that on my 2012 with 40kwh, my SOC drops as soon as I start the Chademo session, and reads far less than the incoming, whilst charging - clearly not actually an issue, as I can see from the charger how much I've taken on board, but baffles my wife sometimes when she's alone .....

This is required to get charging to work over the original 24kWh capacity. If we don't set the % to lower than it actually is, charging will stop as soon as the original 24kWh capacity is full.

**Comment by Toby-TheEnglishMechanic** [comment]
I figured there'd be a good reason for it 😁
We've driven SO much and SO far with this upgrade, thankyou for ALL your amazing work 😁
